pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -96,7 +96,7 @@ func NewFramework(r Registry, plugins *config.Plugins, args []config.PluginConfi
 		// when configured.
 		f.pluginNameToWeightMap[name] = int(pg[name].Weight)
 		if f.pluginNameToWeightMap[name] == 0 {
-			f.pluginNameToWeightMap[name] = 1
+			return nil, fmt.Errorf("plugin %v is not configured with weight", name)
```

Comment:
The [comment](https://github.com/kubernetes/kubernetes/pull/80305#discussion_r304919930) explained it.

Reply:
The comment in code above this says:
```
A weight of zero is not permitted
```
I think returning error is better way of handling the zero weight situation.

Reply:
`A weight of zero is not permitted`, so we need to set it to 1. And we'll add this warning to document, I think it makes sense.